crates/hir-def/src/nameres/collector.rs

```diff
@@ -1223,8 +1225,9 @@ impl DefCollector<'_> {
                             }
                         };
                         let ast_id = ast_id.with_value(ast_adt_id);
+                        let hygiene = Hygiene::new(self.db.upcast(), file_id);
```

Comment:
(Haven't checked) I feel like we should have a hygiene lying around here somewhere already no? (Building hygiene isn't the cheapest from what I remember)

Reply:
All paths but those in attributes seem to be lowered during `ItemTree` construction, which is decoupled from `DefCollector`, so there's no `Hygiene` around here. Moreover, a single `DefCollector` operates on possibly multiple `HirFileId`s, so we can't just cache `Hygiene` in it.

One thing we could try is having something like `HashMap<HirFileId, OnceCell<Hygiene>>` in `DefCollector` and building `Hygiene` on demand. How does that sound?

Reply:
Ye I think adding that to reuse hygienes sounds good for now